src/runtime_src/xrt/device/hal.cpp

```diff
@@ -143,33 +149,9 @@ createHalDevices(hal::device_list& devices, const std::string& dll, unsigned int
   if (!handle)
     throw std::runtime_error("Failed to open HAL driver '" + dll + "'\n" + xrt_core::dlerror());
 
-  typedef unsigned int (* probeFuncType)();
-
-  auto probeFunc = (probeFuncType)xrt_core::dlsym(handle.get(), propeFunc().c_str());
-  if (!probeFunc)
-    return;
-
-  unsigned pmdCount = 0;
-
-  if (count || (count = probeFunc()) || pmdCount) {
-  // Version of HAL
-    typedef unsigned int (* versionFuncType)();
-    auto vFunc = (versionFuncType)xrt_core::dlsym(handle.get(), versionFunc().c_str());
-    auto version = 1;
-    if (vFunc)
-      version = vFunc();
-
-    if (version==1) {
-      throw std::runtime_error("Legacy HAL version " + std::to_string(version) + " not supported");
-    }
-    else if (version==2) {
-      hal2::createDevices(devices,dll,handle.release(),count);
-    }
-    else
-      throw std::runtime_error("HAL version " + std::to_string(version) + " not supported");
-  }
+  hal2::createDevices(devices, dll, handle.release(), count);
```

Comment:
Confused about the handle here?
I think there should be no need to dlopen shim after everything goes through xrt_core::device?